File: LeetCode/1413. Minimum Value to Get Positive Step by Step Sum.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int ans = 0, sum = 0;
        for(auto i : nums)
        {
            sum += i ; 
            ans = min(ans, sum) ;
        }
        return -ans + 1 ;
    }
};

File: LeetCode/1413. Minimum Value to Get Positive Step by Step Sum.java
/***  KNOWLEDGE IS POWER  ***/

class Solution {
    public int minStartValue(int[] nums) {
        int minVal = 0;
        int prefixSum = 0 ;
        for (int n : nums) 
        {
            prefixSum += n;
            minVal = Math.min(minVal, prefixSum);
        }
        return 1 - minVal;
    }
}
